design/keycloak.x/storage-hot-rod.md

```diff
@@ -91,34 +91,44 @@ to the message definition and updating the definition in the Infinispan server a
 
 ### Changing field name/type
 
-It is possible to change field names or types. However, such changes can introduce backward incompatibility that we need to avoid due to no-downtime requirements. We can avoid breaking backward compatibility by doing updates in the following steps (in this example, we have the field `clientTemplateId` in the first object version (`entityVersion = 1`) and we want to replace it with `clientScopeId`; the migration path is following `clientScopeId = "template-" + clientTemplateId`):
+It is possible to change field names or types. However, such changes can introduce backward incompatibility that we need to avoid due to no-downtime requirements. We can avoid breaking backward compatibility by doing updates in the  steps explained below. 
+
+The steps will be illustrated on the following example: let us have a message definition containing field `clientTemplateId` in the first physical version (`entityVersion = 1`). We want to replace it with `clientScopeId`. The migration path is prescribed as `clientScopeId = "template-" + clientTemplateId`).
 
 ---
 
 **NOTE**
 
 In this proposal, we distinguish between two types of version:
 
-* **Version of storage** - This corresponds to the version of source code and message definitions on the Keycloak server side; in other words an incremented storage version means there was a release of Keycloak. In a world where there is no change in Keycloak codebase other than changes in `InfinispanObjectEntity`, `entityVersion` corresponds to Keycloak version.
-* **Version of object** - This refers to the version of object in the storage (Infinispan server); in other words, this says what version of storage originally created the object in the store. The object version can be incremented by reading the old object from the store by newer storage, migrating it, and storing it back into the storage.
+* **Version of storage** - This corresponds to the version of source code and message definitions on the Keycloak server side. Each storage area (clients, realms, groups, etc.) has its own independent storage version that is incremented only when there is a change in the source code or message definition in the corresponding storage area. 
```

Comment:
This spec really should not introduce a new notions that are shared across all the storages. This should be done in the `storage-persistence.md`, and it seems it is the `current store version` defined there.
I take it that we need to invent a better name for `logical object version`.